internal/ethapi/api.go

```diff
@@ -1003,9 +1003,12 @@ func (s *PublicTransactionPoolAPI) GetRawTransactionByHash(ctx context.Context,
 func (s *PublicTransactionPoolAPI) GetTransactionReceipt(hash common.Hash) (map[string]interface{}, error) {
 	tx, blockHash, blockNumber, index := core.GetTransaction(s.b.ChainDb(), hash)
 	if tx == nil {
-		return nil, nil
+		return nil, fmt.Errorf("Missing transaction for hash %s", hash.String())
 	}
```

Comment:
Error's should be lower cased, so multiple may be composed nicely. Also I don't think it's necessary to add the hash to the error, it's pretty obvious based on the function call parameter what's not being found. I think `unknown transaction` is fine, no need to make it fancier.